ee/debug/checkups/binary_directory.go

```diff
@@ -68,10 +71,46 @@ func getBinDir() string {
 	case "darwin":
 		return "/usr/local/kolide-k2"
 	case "linux":
+		if allowedcmd.IsNixOS() {
+			return getBinDirOnNixOS()
+		}
 		return "/usr/local/kolide-k2"
 	case "windows":
 		return "C:\\Program Files\\Kolide\\Launcher-kolide-k2\\bin"
 	}
 
 	return ""
 }
+
+// getBinDirOnNixOS returns the most likely binary directory for NixOS,
+// looking through the nix store to find it.
+func getBinDirOnNixOS() string {
+	// The binary directory on NixOS will look like, e.g.,:
+	// `/nix/store/w8k7h0s9v64gqj60as756gg81c788zkp-kolide-launcher-1.4.4-6-g33c6fd9/bin`
+	matches, err := filepath.Glob("/nix/store/*-kolide-launcher-*/bin")
```

Comment:
Is `os.Executable() ` a viable pattern here? Maybe with `path.EvalSymlinks` ?

Reply:
I thought about that, but didn't know if `os.Executable` would be correct when launcher has auto-updated?

Reply:
Good question. Seems easy to test.

Reply:
i wrote myself a test script. Using `execwrapper.Exec` on macOS, `os.Executable` works as you'd expect:
```go
package main

import (
	"context"
	"flag"
	"fmt"
	"os"
	"path/filepath"

	"github.com/kolide/launcher/pkg/execwrapper"
)

func main() {
	fs := flag.NewFlagSet("forker", flag.ExitOnError)
	var (
		flChild = fs.String("child", "", "fork and call child (if present)")
	)

	if err := fs.Parse(os.Args[1:]); err != nil {
		fmt.Printf("Got error parsing: %s\n", err)
		os.Exit(1)
	}

	p, err := os.Executable()
	if err != nil {
		fmt.Printf("Got error calling os.Executable: %s\n", err)
		os.Exit(1)
	}

	preal, err := filepath.EvalSymlinks(p)
	if err != nil {
		fmt.Printf("Got error calling filepath.EvalSymlinks: %s\n", err)
		os.Exit(1)
	}

	fmt.Printf("Hi! I am:        %s\n", p)
	fmt.Printf("which is really: %s\n", preal)

	ctx := context.Background()

	if *flChild != "" {
		fmt.Printf("I am calling:    %s\n\n", p)

		if err := execwrapper.Exec(ctx, *flChild, []string{""}, os.Environ()); err != nil {
			fmt.Printf("Got error from exec: %s\n", err)
			os.Exit(1)
		}
	}
}
```

```shell
$ go build -o /tmp/f ./test-cmds/fork-os.exec/
$ go run ./test-cmds/fork-os.exec/ --child /tmp/f 
Hi! I am:        /var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec
which is really: /private/var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec
I am calling:    /var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec

Hi! I am:        /tmp/f
which is really: /private/tmp/f
```

Reply:
So, we don't want to use `os.Executable()`, then

Reply:
Hrm. I guess that's true, and this is a checkup so we really want the store contents.